pandas/core/common.py

```diff
@@ -1081,12 +1081,23 @@ def _infer_dtype_from_scalar(val):
     return dtype, val
 
 
-def _maybe_cast_scalar(dtype, value):
-    """ if we a scalar value and are casting to a dtype that needs nan -> NaT
-    conversion
+def _maybe_cast(dtype, value):
```

Comment:
actually why don't you roll this routine into ``_maybe_upcast_putmask`` as its only used there (you can just make it an in-line defined function)